python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+        self.log.debug("Configured Noll indices: %s", self.nollIndices)
+
+        # Deferred import of TARTS to handle cases where it's not in the build
+        try:
+            from tarts import NeuralActiveOpticsSys
+        except ImportError as e:
+            raise ImportError(
+                "TARTS neural network system not available. "
+                "Please ensure TARTS is installed and available in the Python path. "
+                f"Original error: {e}"
+            ) from e
+
+        # TARTS system handles None paths by creating new models with random
+        # weights
+        self.tarts = NeuralActiveOpticsSys(
+            os.path.expandvars(self.config.datasetParamPath),
+            os.path.expandvars(self.config.wavenetPath),
+            os.path.expandvars(self.config.alignetPath),
+            os.path.expandvars(self.config.aggregatornetPath),
+        )
+        self.tarts = self.tarts.eval()
+        # Use configurable crop size instead of hard-coded TARTS value
+        self.cropSize = self.config.cropSize
+        self.device = self.config.device
+
+        self.log.info("Initialized TARTS with dataset params: %s", self.config.datasetParamPath)
+        self.log.debug(
+            "Model paths - wavenet: %s, alignet: %s, aggregatornet: %s",
+            self.config.wavenetPath,
+            self.config.alignetPath,
+            self.config.aggregatornetPath,
+        )
+        self.log.info("Running on device: %s", self.device)
+
+        if self.device == "cpu":
+            self.tarts = self.tarts.cpu()
+
+            # Update device attributes for all sub-models to CPU
+            self.tarts.device_val = torch.device("cpu")
+            for model in [
+                self.tarts.alignnet_model,
+                self.tarts.wavenet_model,
+                self.tarts.aggregatornet_model,
+            ]:
+                if model is not None and hasattr(model, "device_val"):
+                    model.device_val = torch.device("cpu")
+
+            # Update device attributes for the underlying models
+            if hasattr(self.tarts.alignnet_model.alignnet, 'device_val'):
+                self.tarts.alignnet_model.alignnet.device_val = torch.device("cpu")
+            if hasattr(self.tarts.wavenet_model.wavenet, 'device_val'):
+                self.tarts.wavenet_model.wavenet.device_val = torch.device("cpu")
+
+            # Ensure all CNN models within the sub-models are also on CPU
+            if hasattr(self.tarts.alignnet_model.alignnet, 'cnn'):
+                self.tarts.alignnet_model.alignnet.cnn = self.tarts.alignnet_model.alignnet.cnn.cpu()
+            if hasattr(self.tarts.wavenet_model.wavenet, 'cnn'):
+                self.tarts.wavenet_model.wavenet.cnn = self.tarts.wavenet_model.wavenet.cnn.cpu()
+            self.log.debug("Moved all sub-models and CNNs to CPU")
+        else:
+            self.tarts.to("cuda")
+            self.log.debug("Moved models to CUDA")
```

Comment:
Just confirming that in this case you don't need to change the device_vals to cuda? And, if so, just out of curiosity, why not?